misaligned.cpp

```diff
@@ -1,21 +1,36 @@
 #include <iostream>
 #include <assert.h>
+#include <string.h>
+
+using namespace std;
+
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+char previousMinorColor[] = "defaultMinor";
 
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
-        }
+  int i = 0, j = 0;
+  for(i = 0; i < 5; i++) {
+    for(j = 0; j < 5; j++) {
+      std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+
+      int returnMinor = strcmp(previousMinorColor, minorColor[i]);
+      if (0 == returnMinor) {
+        std::cout<<"Error : Minor Strings are equal"<<endl;
+        return i * j;
+      }
+      else {
+        std::cout<<"Succes : Minor Strings are unequal"<<endl;
+      }
+      strcpy(previousMinorColor, minorColor[i]);
     }
-    return i * j;
+  }
+  return i * j;
 }
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
-    std::cout << "All is well (maybe!)\n";
-    return 0;
-}
\ No newline at end of file
+  int result = printColorMap();
+  assert(result == 25);

```

Comment:
Test cases does not verify the content of the color map printed. Test cases can be enhanced to check the content too...